src/fides/api/api/v1/endpoints/generic_overrides.py

```diff
@@ -37,27 +40,47 @@ async def list_dataset_paginated(
     page: Optional[int] = Query(None, ge=1),
     search: Optional[str] = Query(None),
     data_categories: Optional[List[str]] = Query(None),
+    exclude_saas_datasets: Optional[bool] = Query(False),
+    only_unlinked_datasets: Optional[bool] = Query(False),
 ) -> Union[Page[Dataset], List[Dataset]]:
     """
     Get a list of all of the Datasets.
     If any pagination parameters (size or page) are provided, then the response will be paginated
-    & provided filters (search, data_categories) will be applied.
+    & provided filters (search, data_categories, exclude_saas_datasets, only_unlinked_datasets) will be applied.
     Otherwise all Datasets will be returned (this may be a slow operation if there are many datasets,
     so using the pagination parameters is recommended).
     """
-    if page or size:
-        query = select(CtlDataset)
-        filter_params = FilterParams(search=search, data_categories=data_categories)
-        filtered_query = apply_filters_to_query(
-            query=query,
-            search_model=CtlDataset,
-            taxonomy_model=CtlDataset,
-            filter_params=filter_params,
+    if not page and not size:
+        return await list_resource(CtlDataset, db)
```

Comment:
We would need to apply the `exclude_saas_datasets` and `only_unlinked_datasets` filters to the non-paginated response if we wanted to keep backwards compatibility. Did we make a decision to omit this?

Reply:
hmm that's a good point. I can definitely add them in -- however I think the idea was to at some point in the future deprecate the non-paginated response entirely since it has performance issues when there's a big number of elements, and the existing non-paginated endpoint with the filters (`/filters/dataset`) still exists so it _can_ be used if that's still needed. 

What do you think? I think ideally if anyone wants to move from the deprecated endpoint to the new one, they should probably add in the pagination support while they're it?

Reply:
added it in https://github.com/ethyca/fides/pull/5132/commits/071258d5d423d453aecb6decaca8d9e424566f57